### DISCUSSION TITLE
JNoise PerlinNoise is 0

### DISCUSSION THREAD

**Discussion by Flexiik** [Minestom/Minestom#1836] [Help]
I'm trying to create procedural generation in Minestom, but when I use JNoise library, like used in Minestom's documentation, it throws that evaluateNoise is 0:
`
PerlinNoiseGenerator perlinCosine=PerlinNoiseGenerator.newBuilder().setSeed(3301).setInterpolation(Interpolation.COSINE).build();

        instance.setGenerator(unit -> {
            Point start = unit.absoluteStart();
            for (int x = 0; x < unit.size().x(); x++) {
                for (int z = 0; z < unit.size().z(); z++) {
                    Point bottom = start.add(x, 0, z);
                    double height = 100 + perlinCosine.evaluateNoise(x, z) * 20;
                    unit.modifier().fill(bottom, bottom.add(1, 0, 1).withY(height), Block.STONE);
                }
            }
        });
        `

**Comment by Flexiik** [comment]
Nevermind, I just downgraded JNoise and it works fine..

**Comment by CheaterTim** [comment]
for future reference: you should be using `JNoise.newBuilder().perlin()` and scale it down to something like `0.05` using `.scale`

```java
JNoise noise = JNoise.newBuilder().perlin(69, Interpolation.COSINE, FadeFunction.IMPROVED_PERLIN_NOISE).scale(0.05).build();

noise.evalulateNoise(x, y);
```
